### DISCUSSION TITLE
Harfbuzz does not apply GPOS mark positioning for Heebo

### DISCUSSION THREAD

**Discussion by DeependraKhetwal** [harfbuzz/harfbuzz#4724] [General] [answered]
Hi We are getting incorrect rendering for hebrew diacritics for Heebo font (https://fonts.google.com/specimen/Heebo ) . Upon investigation we found out that harfbuzz is not applying GPOS mark positioning for diacritics (even after setting 'mark' and 'mkmk' features explicitly for shaping)  because the font does not contain hebrew script under GPOS  and thus falling back to fallback mark positioning. Is it possible that harfbuzz can use the GPOS mark positioning from DLFT script in case the main script (in this case hebrew) is not present in the font GPOS table

**Comment by behdad** [comment]
This was done to work around Times New Roman not having any Hebrew GPOS rules... :(

**Reply by DeependraKhetwal** [reply]
Okay.Is there a way to enable this behaviour in harfbuzz (apply GPOS mark positioning)  for fonts like Heebo which contain only DLFT script?

**Reply by behdad** [reply]
Not currently. Let me think about it. cc @jfkthame

**Comment by behdad** [comment]
Do we know why that font uses `DFLT`?

**Reply by DeependraKhetwal** [reply]
I am not sure why the Heebo font does not contain hebr script and only DLFT when its a hebrew font primarily.

**Comment by behdad** [comment]
Perhaps we can blacklist the affected Times / Arial etc just for Hebrew and remove the special rule. @jfkthame I'd really appreciate your thoughts here.

**Reply by jfkthame** [reply]
I'd usually be in favor of eliminating "special rules" where possible, in favor of more consistent behavior. The one thing that makes me a bit hesitant in this case -- requiring wider investigation -- is that I wonder how many "affected fonts" like Times there will be. Is this a pattern that has existed across many versions of the TNR (and Arial?) fonts? Is it also found in others that may have based their OpenType tables on what these "standard" fonts contained? We might find ourselves having to blacklist more than just a handful of faces, which gets kinda painful to maintain.

Aside from that concern I'd say try it...

**Reply by behdad** [reply]
Thanks.

The other strategy would be to ignore GPOS for Hebrew only if we get DFLT and the font has more than just DFLT in it.

**Reply by jfkthame** [reply]
Yeah, that seems like it could be a reasonable heuristic. I'd be fine with trying that approach.

**Comment by behdad** [comment]
@DeependraKhetwal Can you give me an example Unicode sequence?

**Reply by DeependraKhetwal** [reply]
וֹהֳ   U+05D5,U+05B9, U+05D4,U+05B3

**Comment by behdad** [comment]
> We are getting incorrect rendering for hebrew diacritics for Heebo font (https://fonts.google.com/specimen/Heebo )

What version are you using? I just downloaded the fonts from that link, and Heebo-Regular.ttf definitely has `hebr` script system in GPOS.

**Reply by DeependraKhetwal** [reply, marked as answer]
@behdad I am using font version 3.001 . I had downloaded that font in January when i was started looking at this issue. It seems now they have updated the font to version 3.100 which contains the
[Heebo-Regular.zip](https://github.com/harfbuzz/harfbuzz/files/15478401/Heebo-Regular.zip)
 hebrew script. Attaching the old font file for reference.

**Reply by DeependraKhetwal** [reply]
I tried with latest version (3.100) of the font file and the diacritics are getting positioned correctly with it :)

**Reply by behdad** [reply]
Thanks for checking. Okay then let's close this, since no one else has ever complained about this.

**Reply by DeependraKhetwal** [reply]
Okay . Thanks for analyzing this.
